### PR TITLE
JInstallerAdapterTemplate - rewrote hardcoded SQL to  object

### PR DESCRIPTION
**Pull request by pe7er**
I've rewrote the hard-coded SQL to $query object in two places.

**Note:** The second SQL query contained an in() with subquery.
Please triple double check my implementation of that query with the $query object using a $subquery.

### CODE REVIEW

File: libraries/cms/installer/adapter/template.php [hunk 1]
```diff
@@ -494,11 +498,16 @@ public function uninstall($id)
...
+		$subquery = $db->getQuery(true)
+			->select('s.id')
+			->from('#__template_styles s')
+			->where($db->quoteName('s.template') . ' = ' . $db->quote(strtolower($name)))
+			->where($db->quoteName('s.client_id') . ' = ' . $db->quote($clientId));
+
+		$query->clear()
+			->update($db->quoteName('#__menu'))
+			->set($db->quoteName('template_style_id') . ' = 0')
+			->where($db->quoteName('template_style_id') . ' in (' . $subquery->__toString() . ')');
```
Comment (pe7er) [inline libraries/cms/installer/adapter/template.php]: Thanks @mbabker! I've changed it to (string) $subquery

**Comment by widmann-it** [comment]
I have tested this item :white_check_mark: successfully on 655b987edf20f4b9439c9ae813ae12c4d16795f4<br /><br />successfuly getestet<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12748">issues.joomla.org/tracker/joomla-cms/12748</a>.</sub>

**Comment by pe7er** [comment]
Thanks for testing Marc!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12748">issues.joomla.org/tracker/joomla-cms/12748</a>.</sub>

**Comment by brianteeman** [comment]
Can you look at resolving the conflicts so that this can be tested please.

**Comment by pe7er** [comment]
My PR was to correct the file **/libraries/cms/installer/adapter/template.php** but I cannot find it in the current staging anymore...

**Comment by wojsmol** [comment]
@pe7er see https://github.com/joomla/joomla-cms/blob/staging/libraries/src/Installer/Adapter/TemplateAdapter.php

**Comment by pe7er** [comment]
Thanks @wojsmol !

**Comment by pe7er** [comment]
Because the code has been moved to another place, I've written a new PR #17923 
And closed this one.

**Pull request closed without merging**